src/cascadia/inc/cppwinrt_utils.h

```diff
@@ -107,6 +107,18 @@ public:                                                        \
 private:                                                       \
     type _##name{ __VA_ARGS__ };
 
+// Use this macro to quick implement both the getter and setter for a color property.
+// This should only be used for color types where there's no logic in the
+// getter/setter beyond just accessing/updating the value.
+// This takes advantage of til::color
+#define GETSET_COLORPROPERTY(name, ...)                                             \
+public:                                                                             \
+    winrt::Windows::UI::Color name() const noexcept { return _##name; }             \
+    void name(const winrt::Windows::UI::Color& value) noexcept { _##name = value; } \
+                                                                                    \
+private:                                                                            \

```

Comment:
is this really a generic-enough helper that it belongs in utils

Reply:
Eh? 🤷‍♂️ Yes, because WinUI::Color should be how we expose colors across layers, and we want to leverage til::color as much as we can? Only 50% confident here haha

Reply:
I _think_ that some of this C++/winrt stuff might magically take care of it for us ...


so what happens if you just use `GETSET_PROPERTY(til::color, Foreground)` or whatever?

There's a trick to c++/winrt, and that trick is that your implementation methods don't actually need to match your signatures at all ;)

Reply:
Okay, I think that this _won't_ automatically work because the type it's looking for is actually a `struct_Windows_UI_Color` and the `detach_from` helper does not allow for type coercion.